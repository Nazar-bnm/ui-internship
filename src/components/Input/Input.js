import React from 'react';
import PropTypes from 'prop-types';

const Input = (props) => {
  return (
    <div className="input-wrapper">
      {props.icon && <i className={props.icon + ' icon'}></i>}
      <input type="text" placeholder="SEARCH" className={props.customClass + ' input'} />
    </div>
  );
};

Input.propTypes = {
  icon: PropTypes.string,
  customClass: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
};

export default Input;
