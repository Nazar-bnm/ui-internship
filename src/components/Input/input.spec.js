import React from 'react';
import { shallow } from 'enzyme';

import Input from './Input';

describe('<Input />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Input />
    );
  });
  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
