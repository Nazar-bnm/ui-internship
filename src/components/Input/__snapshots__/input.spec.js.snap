// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should match snapshot 1`] = `
<div
  className="input"
>
  <input
    className=""
    placeholder=""
    type="text"
  />
</div>
`;
